packages/protobuf/src/private/util-common.ts

```diff
@@ -55,18 +54,39 @@ export function makeUtilCommon(): Omit<Util, "newFieldList" | "initFields"> {
               !(val instanceof sourceField.T)
             ) {
               val = new sourceField.T(val);
+            } else if (
+              sourceField &&
+              sourceField.kind === "scalar" &&
+              sourceField.T === ScalarType.BYTES
+            ) {
+              val = toU8Arr(val);
             }
             t[localName] = { case: sk, value: val };
             break;
           case "scalar":
           case "enum":
-            t[localName] = s[localName];
+            let target = s[localName];
+            if (member.T === ScalarType.BYTES) {
+              target = member.repeated
+                ? (target as ArrayLike<number>[]).map(toU8Arr)
+                : toU8Arr(target);
+            }
+            t[localName] = target;
             break;
           case "map":
             switch (member.V.kind) {
               case "scalar":
               case "enum":
-                Object.assign(t[localName], s[localName]);
+                let target = s[localName];
+                if (member.V.T === ScalarType.BYTES) {
+                  target = Object.fromEntries(
+                    Object.entries(target).map(([k, v]) => [
+                      k,
+                      toU8Arr(v as ArrayLike<number>),
+                    ])
+                  );
+                }
+                Object.assign(t[localName], target);
```

Comment:
We cannot use Object.entries unfortunately. Our target is ES2017, and the method was only added in ES2019.

We should investigate why this didn't raise an error.

Reply:
Apologies, I meant Object.**from**Entries 🙈  Object.entries is fine, it's part of ES2017.